packages/request-node/src/storageUtils.ts

```diff
@@ -29,33 +28,33 @@ export function getEthereumStorage(
   };
 
   // Initializes web3 connection object
-  let provider: HDWalletProvider;
-  if (config.getStorageWeb3ProviderUrl().match('^wss?://.+')) {
-    provider = new HDWalletProvider({
-      mnemonic,
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      providerOrUrl: new Web3WsProvider(config.getStorageWeb3ProviderUrl(), {
-        clientConfig: {
-          keepalive: true,
-          keepaliveInterval: 10000, // ms
-        },
-        // Enable auto reconnection
-        reconnect: {
-          auto: true,
-          delay: 3000, // ms
-          maxAttempts: 5,
-          onTimeout: false,
-        },
-      }),
-    });
+
+  let provider: providers.Provider;
+  const url = config.getStorageWeb3ProviderUrl();
+  if (url.match('^wss?://.+')) {
+    provider = new providers.WebSocketProvider(url);
+    // FIXME previous WS config
+    // {
+    //   clientConfig: {
+    //     keepalive: true,
+    //     keepaliveInterval: 10000, // ms
+    //   },
+    //   // Enable auto reconnection
+    //   reconnect: {
+    //     auto: true,
+    //     delay: 3000, // ms
+    //     maxAttempts: 5,
+    //     onTimeout: false,
+    //   },
+    // }
   } else {
-    provider = new HDWalletProvider(mnemonic, config.getStorageWeb3ProviderUrl());
+    provider = new providers.StaticJsonRpcProvider(url);
```

Comment:
using a StaticJsonRpcProvider allows to reduce the number of calls (eth_chainId for instance)

Reply:
I discovered it with ct. I should have told you so.